receiver.go

```diff
@@ -216,24 +214,14 @@ func (r *Receiver) Listen(ctx context.Context, handler Handler) *ListenerHandle
 	ctx, span := r.startConsumerSpanFromContext(ctx, "sb.Receiver.Listen")
 	defer span.End()
 
-	messages := make(chan *amqp.Message)
-	go r.listenForMessages(ctx, messages)
-	go r.handleMessages(ctx, messages, handler)
```

Comment:
the handleMessages is now folded into the `amqpAdapterHandler` that wraps the user handler